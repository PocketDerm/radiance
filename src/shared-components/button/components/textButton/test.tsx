import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { decorateWithThemeProvider } from 'src/tests/decorateWithThemeProvider';

import { TextButton } from './index';

describe('<TextButton />', () => {
  const DecoratedTextButton = decorateWithThemeProvider(TextButton);

  describe('UI snapshots', () => {
    it('renders without any props', () => {
      const tree = renderer
        .create(<DecoratedTextButton>Button Text</DecoratedTextButton>)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('renders with disabled prop', () => {
      const tree = renderer
        .create(
          <DecoratedTextButton disabled>
            Disabled Button Text
          </DecoratedTextButton>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('onClick callback', () => {
    it('should invoke onClick', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <DecoratedTextButton onClick={spy}>Button Text</DecoratedTextButton>,
      );

      const button = wrapper.find('button');

      button.simulate('click');
      expect(spy).toHaveBeenCalled();
    });

    it('should not be clickable if disabled', () => {
      const spy = jest.fn();
      const wrapper = mount(
        <DecoratedTextButton disabled onClick={spy}>
          Button Text
        </DecoratedTextButton>,
      );

      const button = wrapper.find('button');

      button.simulate('click');
      expect(spy).not.toHaveBeenCalled();
    });
  });
});
