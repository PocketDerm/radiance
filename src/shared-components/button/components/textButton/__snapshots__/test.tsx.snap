// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> UI snapshots renders with disabled prop 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  -webkit-transition: -webkit-transform 350ms;
  -webkit-transition: transform 350ms;
  transition: transform 350ms;
  width: 100%;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.emotion-0 > svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin-right: 1rem;
  margin-top: -5px;
}

.emotion-2 {
  border-color: transparent;
  background-color: transparent;
  color: #c3c0cd;
  cursor: not-allowed;
}

.emotion-2:hover {
  opacity: 0.8;
  background-color: transparent;
}

.emotion-2:active,
.emotion-2:focus {
  outline: none;
}

<button
  className="emotion-2 emotion-3"
  disabled={true}
  onClick={[Function]}
>
  <div
    className="emotion-0 emotion-1"
  >
    Disabled Button Text
  </div>
</button>
`;

exports[`<TextButton /> UI snapshots renders without any props 1`] = `
.emotion-2 {
  border-color: transparent;
  background-color: transparent;
  color: #332e54;
  cursor: pointer;
}

.emotion-2:hover {
  opacity: 0.8;
  background-color: transparent;
}

.emotion-2:active,
.emotion-2:focus {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0;
  -webkit-transition: -webkit-transform 350ms;
  -webkit-transition: transform 350ms;
  transition: transform 350ms;
  width: 100%;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.emotion-0 > svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin-right: 1rem;
  margin-top: -5px;
}

<button
  className="emotion-2 emotion-3"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="emotion-0 emotion-1"
  >
    Button Text
  </div>
</button>
`;
